Not Found: /favicon.ico
Internal Server Error: /admin_panel/rate/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 17, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 76, in rate
    'rate': current_rate.rate,
            ^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'rate'
Not Found: /favicon.ico
Internal Server Error: /admin_panel/parking-history/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 19, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 97, in parking_history
    return render(request, 'admin_panel/parking_history.html', {'parking_history': parking_history})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
           ^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\loader_tags.py", line 295, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\defaulttags.py", line 861, in do_for
    nodelist_loop = parser.parse(
                    ^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\defaulttags.py", line 961, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 481, in parse
    raise self.error(token, e)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 479, in parse
    filter_expression = self.compile_filter(token.contents)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 597, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 698, in __init__
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: Could not parse the remainder: '%else%' from '%else%'
Internal Server Error: /admin_panel/parking-history/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 19, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 97, in parking_history
    return render(request, 'admin_panel/parking_history.html', {'parking_history': parking_history})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    return parser.parse()
           ^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\loader_tags.py", line 295, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\defaulttags.py", line 861, in do_for
    nodelist_loop = parser.parse(
                    ^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 510, in parse
    raise self.error(token, e)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 508, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\defaulttags.py", line 961, in do_if
    nodelist = parser.parse(("elif", "else", "endif"))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 481, in parse
    raise self.error(token, e)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 479, in parse
    filter_expression = self.compile_filter(token.contents)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 597, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\template\base.py", line 698, in __init__
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: Could not parse the remainder: '% else %' from '% else %'
Internal Server Error: /admin_panel/reports/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 125, in reports
    'Plate Number': [parking.user.plates_set.first().plate_number if parking.user.plates_set.exists() else '' for parking in parking_history],
                                                                     ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'UserProfile' object has no attribute 'plates_set'
Internal Server Error: /admin_panel/reports/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 23, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 136, in reports
    'Cost': parking.cost
            ^^^^^^^^^^^^
AttributeError: 'History' object has no attribute 'cost'
Internal Server Error: /admin_panel/reports/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 23, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 144, in reports
    with pd.ExcelWriter(response, engine='xlsxwriter') as writer:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\excel\_xlsxwriter.py", line 197, in __init__
    from xlsxwriter import Workbook
ModuleNotFoundError: No module named 'xlsxwriter'
Internal Server Error: /admin_panel/reports/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 23, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 145, in reports
    df.to_excel(writer, index=False, sheet_name='Parking History')
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\util\_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\core\generic.py", line 2417, in to_excel
    formatter.write(
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 952, in write
    writer._write_cells(
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\excel\_xlsxwriter.py", line 261, in _write_cells
    for cell in cells:
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 890, in get_formatted_cells
    cell.val = self._format_value(cell.val)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 607, in _format_value
    raise ValueError(
ValueError: Excel does not support datetimes with timezones. Please ensure that datetimes are timezone unaware before writing to Excel.
Internal Server Error: /admin_panel/reports/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 144, in reports
    df.to_excel(writer, index=False, sheet_name='Parking History')
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\util\_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\core\generic.py", line 2417, in to_excel
    formatter.write(
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 952, in write
    writer._write_cells(
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\excel\_xlsxwriter.py", line 261, in _write_cells
    for cell in cells:
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 890, in get_formatted_cells
    cell.val = self._format_value(cell.val)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 607, in _format_value
    raise ValueError(
ValueError: Excel does not support datetimes with timezones. Please ensure that datetimes are timezone unaware before writing to Excel.
Internal Server Error: /admin_panel/parking_history/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 89, in parking_history
    print(parking_history)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\query.py", line 379, in __repr__
    return "<%s %r>" % (self.__class__.__name__, data)
                                                 ^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\base.py", line 588, in __repr__
    return "<%s: %s>" % (self.__class__.__name__, self)
                                                  ^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\parkings\models.py", line 36, in __str__
    return f"Parking {self.id} for user {self.user}"
                                         ^^^^^^^^^
AttributeError: 'History' object has no attribute 'user'
Internal Server Error: /admin_panel/parking_history/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 89, in parking_history
    print(parking_history)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\query.py", line 379, in __repr__
    return "<%s %r>" % (self.__class__.__name__, data)
                                                 ^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\base.py", line 588, in __repr__
    return "<%s: %s>" % (self.__class__.__name__, self)
                                                  ^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\parkings\models.py", line 36, in __str__
    return f"Parking {self.id} for user {self.user}"
                                         ^^^^^^^^^
AttributeError: 'History' object has no attribute 'user'
Internal Server Error: /admin_panel/parking_history/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 89, in parking_history
    print(parking_history)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\query.py", line 379, in __repr__
    return "<%s %r>" % (self.__class__.__name__, data)
                                                 ^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\base.py", line 588, in __repr__
    return "<%s: %s>" % (self.__class__.__name__, self)
                                                  ^^^^
TypeError: __str__ returned non-string (type History)
Internal Server Error: /admin_panel/reports/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 115, in reports
    for parking in parking_history:
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\query.py", line 400, in __iter__
    self._fetch_all()
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 1549, in execute_sql
    sql, params = self.as_sql()
                  ^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 736, in as_sql
    extra_select, order_by, group_by = self.pre_sql_setup(
                                       ^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 84, in pre_sql_setup
    self.setup_query(with_col_aliases=with_col_aliases)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 73, in setup_query
    self.select, self.klass_info, self.annotation_col_map = self.get_select(
                                                            ^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 279, in get_select
    related_klass_infos = self.get_related_selections(select, select_mask)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 1367, in get_related_selections
    raise FieldError(
django.core.exceptions.FieldError: Invalid field name(s) given in select_related: 'user'. Choices are: plate
Internal Server Error: /admin_panel/reports/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 120, in reports
    for parking in parking_history:
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\query.py", line 400, in __iter__
    self._fetch_all()
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 1549, in execute_sql
    sql, params = self.as_sql()
                  ^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 736, in as_sql
    extra_select, order_by, group_by = self.pre_sql_setup(
                                       ^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 84, in pre_sql_setup
    self.setup_query(with_col_aliases=with_col_aliases)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 73, in setup_query
    self.select, self.klass_info, self.annotation_col_map = self.get_select(
                                                            ^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 279, in get_select
    related_klass_infos = self.get_related_selections(select, select_mask)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 1367, in get_related_selections
    raise FieldError(
django.core.exceptions.FieldError: Invalid field name(s) given in select_related: 'user'. Choices are: plate
Internal Server Error: /admin_panel/reports/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 120, in reports
    for parking in parking_history:
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\query.py", line 400, in __iter__
    self._fetch_all()
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 1549, in execute_sql
    sql, params = self.as_sql()
                  ^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 736, in as_sql
    extra_select, order_by, group_by = self.pre_sql_setup(
                                       ^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 84, in pre_sql_setup
    self.setup_query(with_col_aliases=with_col_aliases)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 73, in setup_query
    self.select, self.klass_info, self.annotation_col_map = self.get_select(
                                                            ^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 279, in get_select
    related_klass_infos = self.get_related_selections(select, select_mask)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\db\models\sql\compiler.py", line 1367, in get_related_selections
    raise FieldError(
django.core.exceptions.FieldError: Invalid field name(s) given in select_related: 'user'. Choices are: plate
Internal Server Error: /admin_panel/reports/parking-report/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 146, in generate_parking_report
    df.to_excel(writer, sheet_name='Parking History', index=False)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\util\_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\core\generic.py", line 2417, in to_excel
    formatter.write(
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 952, in write
    writer._write_cells(
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\excel\_xlsxwriter.py", line 261, in _write_cells
    for cell in cells:
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 890, in get_formatted_cells
    cell.val = self._format_value(cell.val)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 607, in _format_value
    raise ValueError(
ValueError: Excel does not support datetimes with timezones. Please ensure that datetimes are timezone unaware before writing to Excel.
Internal Server Error: /admin_panel/reports/parking-report/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 146, in generate_parking_report
    df.to_excel(writer, sheet_name='Parking History', index=False)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\util\_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\core\generic.py", line 2417, in to_excel
    formatter.write(
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 952, in write
    writer._write_cells(
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\excel\_xlsxwriter.py", line 261, in _write_cells
    for cell in cells:
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 890, in get_formatted_cells
    cell.val = self._format_value(cell.val)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 607, in _format_value
    raise ValueError(
ValueError: Excel does not support datetimes with timezones. Please ensure that datetimes are timezone unaware before writing to Excel.
Internal Server Error: /admin_panel/reports/parking-report/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 22, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 146, in generate_parking_report
    df.to_excel(writer, sheet_name='Parking History', index=False)
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\util\_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\core\generic.py", line 2417, in to_excel
    formatter.write(
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 952, in write
    writer._write_cells(
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\excel\_xlsxwriter.py", line 261, in _write_cells
    for cell in cells:
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 890, in get_formatted_cells
    cell.val = self._format_value(cell.val)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 607, in _format_value
    raise ValueError(
ValueError: Excel does not support datetimes with timezones. Please ensure that datetimes are timezone unaware before writing to Excel.
Internal Server Error: /admin_panel/reports/parking-report/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 26, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 149, in generate_parking_report
    df.to_excel(writer, index=False, sheet_name='Parking History')
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\util\_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\core\generic.py", line 2417, in to_excel
    formatter.write(
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 952, in write
    writer._write_cells(
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\excel\_openpyxl.py", line 486, in _write_cells
    for cell in cells:
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 890, in get_formatted_cells
    cell.val = self._format_value(cell.val)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\pandas\io\formats\excel.py", line 607, in _format_value
    raise ValueError(
ValueError: Excel does not support datetimes with timezones. Please ensure that datetimes are timezone unaware before writing to Excel.
Internal Server Error: /admin_panel/reports/parking-report/
Traceback (most recent call last):
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "E:\Users\mal4ishka\AppData\Local\pypoetry\Cache\virtualenvs\parkeasy-eFmvKdua-py3.12\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 26, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Python\GoIT\Module 3\ParkEasy\ParkEasy\admin_panel\views.py", line 137, in generate_parking_report
    'Parking End': parking.parking_end.replace(tzinfo=None),
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'replace'
